src/video/scale2x.h

```diff
@@ -65,7 +65,7 @@ typedef uint32_t scale2x_uint32;
 /**
  * Align a pointer.
  */
-static void* scale2x_align_ptr(const void* ptr)
+static inline void* scale2x_align_ptr(const void* ptr)
```

Comment:
that commit make the build fail : 
https://travis-ci.org/OpenDUNE/OpenDUNE/builds/415198858

Reply:
I can't reproduce this and I was very puzzled what was going on. Adding "inline" to functions in header file isn't a first for me but that it would break a build, yes.

I suspect that since there's no mention of which version of the C standard in CFLAGS, each compiler will select a different one, that's why we see this discrepancy between environments.

I see that at various place in the code, we have similar constructs:

```
#ifdef _MSC_VER
#define inline __inline
#define restrict __restrict
#else /* _MSC_VER */
#define inline __inline__
#define restrict __restrict__
#endif /* _MSC_VER */
```

I could try to add that at the beginning of "scale2x.h" which might work... or not because if previously in a compilation unit we try to "#define include" again, it will break.

We also have another attempt to do something similar in "config.lib":

```
CFLAGS="$CFLAGS -Dinline=__inline__"
```

Because of all these reasons, I think there's 2 ways to solve this problem once and for all:

* Force a C standard in CFLAGS which have a definition for "inline".
* Remove all attempt in the code base to #define include (and friends) and create a new header file, let's call it "compiler.h" and add all these defines here. All #define sites we removed now should #include "compiler.h" instead. This new header will be protected by "include guards" so if a compilation unit #include "compiler.h" more than once, it will not break as we will not "#define inline" more than once.

The second approach looks better to me long term as if other compiler specific idiosyncrasies do popup, they can be added to that file.

Reply:
https://github.com/OpenDUNE/OpenDUNE/wiki/Coding-Style
inline is not C89